framework/src/modules/chain/submodules/blocks.js

```diff
@@ -148,29 +148,11 @@ class Blocks {
 	// eslint-disable-next-line class-methods-use-this
 	async onNewBlock(block) {
 		if (components && components.cache && components.cache.isReady()) {
```

Comment:
I guess this condition should also be in http_api module.

Reply:
If the check is not here the publish never happens so the subscriber will never fire it's callback

Reply:
Due to the other change this change was also made for validating the status of the cache